### DISCUSSION TITLE
Azure OpenAI covering

### DISCUSSION THREAD

**Discussion by dhuynh95** [outlines-dev/outlines#404] [Feature requests]
Hi everyone,
I love the project and am using outlines to turn GPT4 in a reliable zero shot classifier.
I have to use Azure OpenAI for compliance reasons.

Could you tell me how to modify / use outlines to leverage Azure OpenAI instead of OpenAI? I am using OpenAI 1.37, and using this code works for me to consume Azure OpenAI (from [official docs](https://learn.microsoft.com/en-us/azure/ai-services/openai/how-to/migration?tabs=python-new%2Cdalle-fix)):
```python
import os
from openai import AzureOpenAI

client = AzureOpenAI(
  azure_endpoint = os.getenv("AZURE_OPENAI_ENDPOINT"), 
  api_key=os.getenv("AZURE_OPENAI_KEY"),  
  api_version="2023-05-15"
)

response = client.chat.completions.create(
    model="gpt-35-turbo", # model = "deployment_name".
    messages=[
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": "Does Azure OpenAI support customer managed keys?"},
        {"role": "assistant", "content": "Yes, customer managed keys are supported by Azure OpenAI."},
        {"role": "user", "content": "Do other Azure AI services support this too?"}
    ]
)

print(response.choices[0].message.content)
```

**Comment by rlouf** [comment]
Hi,

Given the APIs are identical, I think it would make sense to either have a keyword argument when initialising `OpenAI`, or overloading its `__init__` function. We can then add a `models.azure_openai` alias. I would be more than happy to review a PR on that.

**Reply by dhuynh95** [reply]
That would be great! I think it would be a great addition to help all users having to work in enterprise environments where OpenAI is not allowed, but Azure OpenAI is!
Would be glad to give a hand if needed but I guess it would be faster for you to do.

Looking forward to that feature!

**Reply by rlouf** [reply]
Faster for who? 😄 I'll take a look when I get to that part of the codebase, we still have a PR open related to OpenAI.
